File: backend/app/models/education.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

# Request Models
class EducationCreate(BaseModel):
    institution: str
    location: str
    degree: str
    gpa: Optional[float] = None
    max_gpa: Optional[float] = None
    start_date: str
    end_date: str  # Can be "Present" or date
    courses: Optional[List[str]] = None

class EducationUpdate(BaseModel):
    institution: Optional[str] = None
    location: Optional[str] = None
    degree: Optional[str] = None
    gpa: Optional[float] = None
    max_gpa: Optional[float] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    courses: Optional[List[str]] = None

# Response Models
class EducationResponse(BaseModel):
    id: str
    user_id: str
    institution: str
    location: str
    degree: str
    gpa: Optional[float] = None
    max_gpa: Optional[float] = None
    start_date: str
    end_date: str
    courses: Optional[List[str]] = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/api/education.ts
import api from './axios'

export interface EducationResponse {
  id: number
  user_id: number
  institution: string
  location: string
  degree: string
  gpa: string | number | null // Can be string "7.84 / 10.00" or number
  max_gpa?: number | null // Separate max_gpa field if backend returns it
  start_date: string
  end_date: string
  courses: string[] | null
  created_at: string
  updated_at: string
}

export interface EducationCreate {
  institution: string
  location: string
  degree: string
  gpa?: number | null
  max_gpa?: number | null
  start_date: string
  end_date: string
  courses?: string[] | null
}

export interface EducationUpdate {
  institution?: string
  location?: string
  degree?: string
  gpa?: number | null
  max_gpa?: number | null
  start_date?: string
  end_date?: string
  courses?: string[] | null
}

export const educationApi = {
  getEducation: async (): Promise<EducationResponse[]> => {
    const response = await api.get<EducationResponse[]>('/api/education/')
    return response.data
  },

  getEducationById: async (educationId: number): Promise<EducationResponse> => {
    const response = await api.get<EducationResponse>(`/api/education/${educationId}`)
    return response.data
  },

  createEducation: async (data: EducationCreate): Promise<EducationResponse> => {
    const response = await api.post<EducationResponse>('/api/education/', data)
    return response.data
  },

  updateEducation: async (educationId: number, data: EducationUpdate): Promise<EducationResponse> => {
    const response = await api.put<EducationResponse>(`/api/education/${educationId}`, data)
    return response.data
  },

  deleteEducation: async (educationId: number): Promise<void> => {
    await api.delete(`/api/education/${educationId}`)
  },
}
